chapters/packages.tex

```diff
@@ -321,3 +321,135 @@ \section{External Resources}\label{external-resources}
+\item One file for each supported language with the name \filename{$\mathit{LibraryName}$.$\mathit{language}$.po}, as the resource \filename{modelica://$\mathit{LibraryName}$/Resources/Language/$\mathit{LibraryName}$.$\mathit{language}$.po}.
+This file is a copy of the associated template file, but extended with the translations in the specified language.
+The pattern \filename{$\mathit{language}$} stands for the ISO 639-1 language code, e.g., \filename{de} or \filename{en}.
+\end{enumerate}
+
+The detailed format of these files is described in \textcite{GettextManual}.
+For Modelica translations, only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used, meaning that a translation entry looks like this:
+\begin{lstlisting}
+#: $\mathit{filename}$:$\mathit{lineNumber}$
+# no-c-format
+msgctxt $\mathit{messageContext}$
+msgid $\mathit{messageIdentifier}$
+msgstr $\mathit{messageTranslation}$
+\end{lstlisting}
+
+The use of \lstinline!no-c-format! ensures that translation tools will not parse \lstinline!"%class"! as the format specifier \lstinline!%c! followed by \emph{lass}.
+
+\begin{nonnormative}
+In the remainder of this section, several facts about the gettext specification are interleaved non-normatively for easy access to some of the gettext basics.
+Always refer to the external gettext specification for full detail or in case of doubt.
+
+All text strings are in double quotes and encoded with UTF-8 characters.
+Comments start with an \lstinline!#! and are continued until the end of line.
+Spaces outside strings are ignored and used as separators.
+
+The files consist of a header and a body.
+The header is marked with an empty \lstinline!msgid! and looks like this:
+\begin{lstlisting}
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2019-03-15 10:52+0100\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+\end{lstlisting}
+All general terms in the header should be replaced by specific information.
+
+Following the header, there is one translation entry for each string to be translated.
+It can start with an optional comment describing the location (file name and line number) of the text to translate.
+Multiple occurences of the same string can be listed here, separated by space.
+\end{nonnormative}
+
+The $\mathit{messageContext}$ following the keyword \lstinline!msgctxt! shall be the full name of the Modelica class (e.g., \lstinline!"Modelica.Blocks.Math.Sin"!) where the text appears.
```

Comment:
It's not clear to me if the context is optional or not. It could potentially be useful to omit the context if a string is used multiple times in different contexts, but that's perhaps not a very common use case.

Reply:
The original idea was to have it optional, and I guess that is also the case in the prototype implementation of SimulationX. @ooer Can you confirm?

Reply:
Isn't this the kind of question where we should try to stick to common gettext practices?

Reply:
Having the messageContext always present and defined as identifier of the surrounding class would ease the tool implementation of some library operations like e.g. renaming/moving of classes and sub packages with recognition of the translation. Conversion scripts (class part) can also applied to the translated files then.

Reply:
Yes, mandatory class context was the conclusion.